synapse/handlers/room.py

```diff
@@ -521,8 +521,24 @@ def send_membership_event(
                 action = "remote_join"
         elif event.membership == Membership.LEAVE:
             is_host_in_room = self.is_host_in_room(context.current_state)
+
             if not is_host_in_room:
-                action = "remote_reject"
+                # perhaps we've been invited
+                inviter = self.get_inviter(target_user.to_string(), context.current_state)
+                if not inviter:
+                    raise SynapseError(404, "Not a known room")
+
+                if inviter.domain == self.server_name:
```

Comment:
I just lifted the code from `do_remotely_reject_invite`. I guess your way is better

<rant> if anybody had bothered to document what self.hs actually is, this would have been somewhat easier to figure out. </rant>